packages/mobx/__tests__/v5/base/make-observable.ts

```diff
@@ -552,3 +569,884 @@ test("makeObservable respects options.name #2614'", () => {
+    expect(isAction(child.action)).toBe(true)
+
+    expect(child.hasOwnProperty("action")).toBe(false)
+
+    expect(Parent.prototype.action()).toBe("parent")
+    expect(Child.prototype.action()).toBe("child of parent")
+    expect(child.action()).toBe("child of parent")
+
+    // Action bound
+    expect(isAction(Parent.prototype["action.bound"])).toBe(true)
+    expect(isAction(Child.prototype["action.bound"])).toBe(true)
+    expect(isAction(child["action.bound"])).toBe(true)
+
+    expect(child.hasOwnProperty("action.bound")).toBe(true)
+
+    expect(Parent.prototype["action.bound"]()).toBe("parent")
+    expect(Child.prototype["action.bound"]()).toBe("child of parent")
+    expect(child["action.bound"]()).toBe("child of parent")
+
+    // Computed
+    expect(isComputedProp(child, "computed")).toBe(true)
+    expect(child.computed).toBe("child")
+
+    // Flow
+    expect(isFlow(Parent.prototype.flow)).toBe(true)
+    expect(isFlow(Child.prototype.flow)).toBe(true)
+    expect(isFlow(child.flow)).toBe(true)
+
+    expect(child.hasOwnProperty("flow")).toBe(false)
+
+    expect(await Parent.prototype.flow()).toBe("parent")
+    expect(await Child.prototype.flow()).toBe("child of parent")
+    expect(await child["action.bound"]()).toBe("child of parent")
+})
+
+test("subclass - cannot re-annotate", () => {
+    class Parent {
+        observable = 1
+        constructor() {
+            makeObservable(this, {
+                action: action,
+                observable: observable,
+                actionBound: action.bound,
+                flow: flow,
+                computed: computed
+            })
+        }
+        action() {}
+        actionBound() {}
+        *flow() {}
+        get computed() {
+            return this
+        }
+    }
+
+    class ChildAction extends Parent {
+        constructor() {
+            super()
+            makeObservable(this, {
+                action: action
+            })
+        }
+        action() {}
+    }
+
+    class ChildActionBound extends Parent {
+        constructor() {
+            super()
+            makeObservable(this, {
+                actionBound: action.bound
+            })
+        }
+        actionBound() {}
+    }
+
+    class ChildFlow extends Parent {
+        constructor() {
+            super()
+            makeObservable(this, {
+                flow: flow
+            })
+        }
+        *flow() {}
+    }
+
+    class ChildObservable extends Parent {
+        constructor() {
+            super()
+            this.observable = 2
+            makeObservable(this, {
+                observable: observable
+            })
+        }
+    }
+
+    class ChildComputed extends Parent {
+        constructor() {
+            super()
+            makeObservable(this, {
+                computed: computed
+            })
+        }
+        get computed() {
+            return this
+        }
+    }
+
+    expect(() => new ChildAction()).toThrow(/^\[MobX\] Cannot apply/)
```

Comment:
There is quite a lot of these assertions and I was changing the messages a lot over time, so I had to go with something that just matches the corrent error, but doesn't depend too much on exact wording.
Secondly the messages contain dynamic info like random object name etc so just assembling the regular got a bit convoluted and I wanted to keep the tests simple and focused.
It can be improved, but I would wait a bit until the impl settles, I don't want to change it tomorrow again because of some spelling error, misleading info or something :D

> NVM, it is on 998

I suppose this was ment for *"I think a test case where @override is missing would be good to have"*

Reply:
true, tests verifying every last letter causes just as much trouble as it solves :) Lets leave it like this.